sql: don't mark subqueries as aggregated

madelynnblue opened the pull request:
Previously, if a subquery used an aggregation function it was marked as
aggregated. This fixes a case in sqllogic select1.test.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5794)

<!-- Reviewable:end -->

RaduBerinde commented:
LGTM

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[sql/testdata/aggregate, line 617 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5794#-KEH1m38wKnuy7yVsOAl:-KEH1m39b1L0sz3ZC-aq:2024682908) ([raw file](https://github.com/cockroachdb/cockroach/blob/b88bf3536747dcde89dbc801d9caaa4bda98fcf0/sql/testdata/aggregate#L617)):*
"GROUP BY requirement" sounds a bit off to me, maybe "don't trigger aggregation."

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5794#-:-KEH1t3Fgmih6L94cpXx:676834233)*
<!-- Sent from Reviewable.io -->

madelynnblue committed:
sql: don't mark subqueries as aggregated

Previously, if a subquery used an aggregation function it was marked as
aggregated. This fixes a case in sqllogic select1.test.

mjibson commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/testdata/aggregate, line 617 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5794#-KEH1m38wKnuy7yVsOAl:-KEI2WrwZmSP9loDflcE:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/b88bf3536747dcde89dbc801d9caaa4bda98fcf0/sql/testdata/aggregate#L617)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5794)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5794)*
<!-- Sent from Reviewable.io -->
